### ISSUE TITLE
ML/AI product experiments

### ISSUE THREAD

**Issue by y0ur1kopf** [anytypeio/roadmap#57]

**Comment by vg16855** [comment]
Hard no. This is just bloat. People who want to use AI tools will use their own. Development resources are better spent on features that help people actually think.

**Comment by werererer** [comment]
Hard yes, only because AI currently has a lot of privacy related negatives, doesn't mean it will always be like that.
if modularly implemented, i think it is great, even if it is "only statistics".
I mean search engine is just an algorithm right, and still people rely on it. This is just search engine++

**Comment by vanyatwo** [comment]
Double hard yes - we should be able to connect Anytype database to own LLMs and train our own digital twins.

**Comment by alexkkoo93** [comment]
Triple hard yes. I'm actually really liking Notion's native AI and the ability to search through all my notes within the app so seamlessly. 

Having this option for AnyType and knowing my data is safe would be a massive W

**Comment by flypenguin** [comment]
Gooood no please. Really. "Experiments" ... that product is by far not polished enough to sink resources in weird "AI experiments". In if you simply do another chatbot in the end – gooooood no.

Good search: Okay. That's a concrete use case, which might be _solved_ using AI.

_Starting_ with AI (like "hmmmm ... what could we do with AI in Anytype?") -- **please** no. Let anyone who wants "AI" make a suggestion which problem that apparently solves.

**Comment by robotflaqd** [comment]
Very hard no (at this point). There are better and more important things to focus on developing at this point than an LLM. There is a topic on the community forum that sums it up real nicely (https://community.anytype.io/t/roadmap-for-local-ai-llm-intergration/22881). This is either going to be half baked or it's going to take up all the time that is better spent developing other features that are highly voted on the feature request page in the community (spoiler: LLM is not one of them). Come back to this only after the app comes out of beta.

**Comment by alexkkoo93** [comment]
> Gooood no please. Really. "Experiments" ... that product is by far not polished enough to sink resources in weird "AI experiments". In if you simply do another chatbot in the end – gooooood no.
> 
> Good search: Okay. That's a concrete use case, which might be _solved_ using AI.
> 
> _Starting_ with AI (like "hmmmm ... what could we do with AI in Anytype?") -- **please** no. Let anyone who wants "AI" make a suggestion which problem that apparently solves.

I'm confused. Are you for or against AI developments? I'm not sure what "Good no" means vs. "Good search" lol. 

My suggestion is for AI to employ RAG the same way Notion does. That is a problem that I feel only notion sufficiently solves

**Comment by flypenguin** [comment]
> I'm confused. Are you for or against AI developments? I'm not sure what "Good no" means vs. "Good search" lol.

oh, true. to be exceptionally clear: F**K AI!!! 😆 . Reasoning: Having AI "because AI!!" is probably nice to hunt for VC money, it does not solve any real-world problem. "Good search" is a problem, which _might_ be solved using AI, and in the end who cares how it was solved – as long as it _is_ solved.

I just (literally a couple of minutes ago) posted a [rather lengthy post](https://community.anytype.io/t/better-keyboard-handling-more-standard-or-expected-behavior/24513) in the forums on what _I_ really, really would like to see for Anytype soon – that is smooth, nice, intuitive behavior and really good keyboard support. To me, "AI" would be something I would actively "oppose" (to be clear: I have absolutely nothing to say here, I just start liking anytype more and more each day, so I start voicing my opinions here 😉) since in my experience it usually takes a ton of resources and delivery incredibly sub-par results.

Also I prefer to operate on tangible goals, and "having AI" ... well, very tangible ;)

**Comment by robotflaqd** [comment]
AI shouldn't even be considered before formulas, tabs, linked/synced blocks are added to Anytype.

**Comment by silpol** [comment]
to me, having 'AI just for sake of AI' is rubber-stamping VC funding applications of last years.

what could be positive examples (these are **examples**!):
- having constantly-learning translation engine, which is used offline (there are others, but this one will learn with your own focus and traits)
- user's own tools to deal with images and videos, from classification to small editing
- more long-tail actual cases ?

**Comment by ebanDev** [comment]
> AI shouldn't even be considered before formulas, tabs, linked/synced blocks are added to Anytype.

I totally agree, furthermore when features as "simple" (putting quotes because I dont mean to disrespect the devs who are doing an amazing work <3)  as ordering tags for example are not available

**Comment by vanyatwo** [comment]
Self-hosted notion alternative with self-hosted LLM integration could bring a lot of devs into the project.

**Comment by Heckinchonkeires** [comment]
I think I'm mostly going to echo things already said here and in the forum, but I feel strongly enough about this to want to add my voice. I want to assume in good faith that this is on the roadmap primarily to placate those that have bought into the genAI hype and are asking genAI be included for the sake of it. That being the case: in my mind this issue should be moved to the backlog and renamed to something like "Generative AI experiments", and "AI assistant" should be removed from the list of highly requested features in the docs. If it's important to explore machine learning in general and outside the context of genAI, then that could be a new issue on the roadmap.

I wouldn't say that genAI should never be a part of Anytype under any circumstances. However, it needs to be approached as an answer to the question "How can Anytype solve [problem]?" and not as a question itself: "How can Anytype include generative AI?" "AI assistant" as a highly requested (read: planned) feature in the docs and this issue's position on the roadmap communicate that Any is approaching genAI as a question. More often than not (at least from what I've seen), that approach leads to throwing genAI at imaginary problems, or at problems that could otherwise be solved in ways that are better—and, importantly, less costly—from both a business perspective and an environmental perspective.

**Comment by robotflaqd** [comment]
I love how this has been pushed to the backlog. There are many other things that Anytype need and that would take up R&D before AI is needed. 

Would love to see AI implemented for Anytype, but definitely not now as I'd love to see many other things get developed _first_. Maybe revisit this once the Anytype passes version 1.0.

**Comment by Xyz00777** [comment]
I think an integration would be aweseome.
BUT the question is what how and when, because I think an integration as something like RAG with asking questions and openai api connection so a person can use his own llm with openai api or chatgpt or anything else, than in the next step I would see the possibility to ask him to write text like cowritting and MAYBE after that ship it as option to activate with a small llm to run local and some other options. Maybe also before implementing an own rag, use technology's who are already there and just give an api to connect.
Also an possibility should be to provide just an MCP (Model Context Protocol) API like the standard Claude has published a while ago.

**Comment by Varming73** [comment]
I dont need yet another ChatGPT inside a different product that has nothing to do with the product itself (which seems to be the case with Capacities). In Obsidian on the other hand I can chat against my notes and have the AI help me find old notes and ideas which is pretty powerful. I honestly dont need an LLM to write my notes for me, but having it tagging my notes consistently would be pretty nice. 

I would also love to be able to just speak my notes/ideas and have them transcribed into Anytype.

So no hard yes/no's from me, just make sure it actually has a real function and is not just glued on to be able to say it's an AI tool please.

**Comment by naught101** [comment]
Please, if this happens, make it optional and opt-in.

Anything non-optional would make Anytype a non-starter for me, and I suspect many others. Anything tied to external AI services would also presumably be a security risk that would prevent many users and organisations from using Anytype.

**Comment by Xyz00777** [comment]
something like that in an application who can be self hosted is/should normaly every time be opt in because normaly you provide your own api keys so it can working 😅

**Comment by BlackRockCity** [comment]
To all the naysayers... install Ollama with Starling-7B LLM and tell me that its not useful.

**Comment by tobyatgithub** [comment]
My 2 cents:

**No notion ai style of ai. Yes for allowing AI API.**

(I recently migrated from notion to obsidian to anytype.)

I have used notion AI, paid for it, didn't like it, and barely use it.
I then use obsidian with AI (a bit workaround). I first edit in obsidian, and then I use cursor to open my vault (everything in markdown), and then use the cursor chat with all my .md as my knowledge base. Absolutely love it. Very handy and HELPS a lot on my brainstorming and thinking.

That being said, I personally don't like and won't expect anytype to have a very integrated AI tool like notion AI. I think it is a waste of time (given how fast AI is developing🤔)

Ideally, I hope anytype can have a "chat"-ish think that can see all the markdown as the knowledge base, and users can pick whatever ai product they wanna connect to (API).

Right now my plan of using anytype with AI is... export my markdowns and open it via cursor😂

Best regards,

**Comment by TaylorFinklea** [comment]
The problem with markdown export is you lose all relationships and the resulting files aren't that useful for AI. I exported as JSON and it was a bit better but too noisy, with settings like the icons and the colors and things like that. 

Ideally, we would be able to export only the notes and objects themselves and their relations rather than everything that Anytype uses. 

When I was using logseq before, this worked very well with my local AI application Msty. Being able to chat with your notes in the Anytype app doesn't seem worth the effort, but being able to export your data in a way that is easily in indigestible by AI would be a huge benefit. Even better if the exports were schedulable.

**Comment by BlackRockCity** [comment]
>  Being able to chat with your notes in the Anytype app doesn't seem worth the effort, but being able to export your data in a way that is easily in indigestible by AI would be a huge benefit. Even better if the exports were schedulable.

A cron job could automate access to an Anytype API. Since the ChatGPT API is increasingly being used by LLMs, integrating it could be highly useful. This would allow users to interact with their Anytype data through an external application like Ollama or a future built-in chat feature, if Anytype decides to implement one.

An API, along with a Linux/Unix CLI tool, would greatly enhance Anytype’s functionality, though it would likely be a lower-priority feature for now.

Potential Benefits

✅ Smart Search & Summarization
	•	Allow users to ask questions about their notes and receive AI-generated answers.
	•	Generate summaries of long notes automatically.

✅ Content Enhancement
	•	Rewrite, clarify, or expand on notes.
	•	Generate action items, highlights, or key takeaways.

✅ Automated Tagging & Organization
	•	Categorize notes based on context.
	•	Suggest relationships between notes.

✅ Scheduled Insights & Reminders
	•	Generate daily/weekly recaps of recent notes.
	•	Remind users of forgotten or relevant notes.

✅ Chat-Based Note Interaction
	•	Users could interact conversationally with their notes instead of just searching manually.
	•	Potential integration with external applications (e.g., personal knowledge management systems).

Edit: I now see this in the .45 release notes... 

> "Our long-awaited API is beginning to take shape! The first step is Raycast integration, allowing you to create, read, and delete Spaces, Objects, and Types, as well as search across them, all from Raycast on macOS.
> While this initial version focuses on a basic set of features, we’re laying the groundwork for an open API that will eventually support bulk import and export, NoCode tools, and more integrations. We’re excited to see what you’ll build!"

**Comment by smellythief** [comment]
I imagine people saying "hard no" either must be imagining it being done poorly or don't want it to delay more basic features. But being able to query your own notes with natural language is shortly going to be table stakes for any notes/database app, and right now would offer huge benefits. If the competition has this but Anytype continues to lack it, that's going to suck after a while. Atm I'd rather have tags as objects though.

**Comment by flypenguin** [comment]
@smellythief Two thoughts:

* As I said [before](https://github.com/anyproto/roadmap/issues/57#issuecomment-2407803337) – what you describe is a very concrete and – I agree – potentially really useful feature, not an "AI experiment".
* The basic text editor has still a lot (!) of room to improve, let's say. Maybe focus on this first ;)

So in generel at least I am not against what _you_ describe, I would just like basic text editing to be pleasant first, and then focus on clear features, not some weird vague "experiments".

**Comment by thomaslule** [comment]
Generative AI comes at such an awful environmental and social price (please read about the data workers manually training AIs in the global south), I dont want to be involved in this mess 🫤

**Comment by albas99** [comment]
I think there are pros and cons to this, ultimately I think similar to how code editors (the classical ones, not the AI ones like cursor and windsurf) give users the ultimately choice to either integrate AI or not into their workflow, users here should also be given the choice, an example is how Zed implements it AI integration or how VS code does with the choice to add an AI extension or not.

**Comment by hayleylabrie** [comment]
> Please, if this happens, make it optional and opt-in.
> 
> Anything non-optional would make Anytype a non-starter for me, and I suspect many others. Anything tied to external AI services would also presumably be a security risk that would prevent many users and organisations from using Anytype.

Here to echo this sentiment. _If_ this is on the table, and IMO it should _not_ be, it needs to be optional, opt-in, and free of pop-ups advertising the feature every time the app is opened.

I stopped using Notion _specifically_ because of the bloated and largely useless AI integration, and I am not alone in that. Outside of a few very narrow use-cases, genAI is a marketing tactic for vibe coders and venture capitalists. PLEASE keep that stuff out of an otherwise very promising open-source project.

**Comment by TaylorFinklea** [comment]
Honestly after the exceptional release of native MCP support, I don't see this being relevant anymore. The MCP server lets me integrate Anytype into any of my LLM interfaces that support it, and that's where I would rather interact with it anyway. Lets those who want to use AI use it, and those who don't get the same Anytype experience.

**Comment by trymeouteh** [comment]
Please considee integrating Ollama which is a private AI solution where you download and self host the LLMs and can choose from many LLMs

**Comment by khols** [comment]
please dont add AI, or at least give us old fashioned folk the option to stay analogue and remove all AI integration from our spaces. I do not want any interaction with the robots

**Comment by mzyxnuel** [comment]
+1, but should be optional

**Comment by trymeouteh** [comment]
> Please considee integrating Ollama which is a private AI solution where you download and self host the LLMs and can choose from many LLMs

I agree AI features should be optional and disabled by default.

**Comment by zhinn0** [comment]
Privacy-first, self-hosted forward is part of the promise of Anytype. I think LLMs can be wonderfully useful but I want, and believe many if not most of the Anytype crowd wants, any AI/LLM integration to be completely opt-in and allow me to use my preferred model(s), whether locally-hosted or via a provider.

Obsidian's API is a good example, as it has made it very easy for the community to create assorted plugins to enable AI use. I'm generally more in favor of this as we can select differing levels of integration and capabilities.

@BlackRockCity had a great list of what would be ideally available in AI integration (assuming the Anytype team owned this instead of taking a community plugins route)

> ✅ Smart Search & Summarization • Allow users to ask questions about their notes and receive AI-generated answers. • Generate summaries of long notes automatically.
> 
> ✅ Content Enhancement • Rewrite, clarify, or expand on notes. • Generate action items, highlights, or key takeaways.
> 
> ✅ Automated Tagging & Organization • Categorize notes based on context. • Suggest relationships between notes.
> 
> ✅ Scheduled Insights & Reminders • Generate daily/weekly recaps of recent notes. • Remind users of forgotten or relevant notes.
> 
> ✅ Chat-Based Note Interaction • Users could interact conversationally with their notes instead of just searching manually. • Potential integration with external applications (e.g., personal knowledge management systems).

**Comment by TomTom101** [comment]
> AI should augment Anytype blablabla

@cYberOptiX Is this the same kind of ai generated crap as on your webpage?
